src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -351,34 +350,37 @@ private boolean isValidBidder(String bidder, Map<String, String> aliases) {
      * - bidrequest.user.ext.data, bidrequest.app.ext.data and bidrequest.site.ext.data will be removed for bidders
      * that don't have first party data allowed.
      */
-    private List<BidderRequest> makeBidderRequests(List<String> bidders, ExtBidRequest requestExt,
-                                                   BidRequest bidRequest, Map<String, String> uidsBody,
+    private List<BidderRequest> makeBidderRequests(List<String> bidders, Map<String, String> aliases,
+                                                   BidRequest bidRequest, ExtBidRequest requestExt,
                                                    UidsCookie uidsCookie, ExtUser extUser, ExtRegs extRegs,
-                                                   Map<String, String> aliases, List<Imp> imps,
-                                                   Map<Integer, Boolean> vendorsToGdpr) {
+                                                   List<Imp> imps, Map<Integer, Boolean> vendorsToGdpr) {
+
+        final Map<String, String> uidsBody = uidsFromBody(extUser);
+
+        final Regs regs = bidRequest.getRegs();
+        final boolean coppaMasking = isCoppaMaskingRequired(regs);
```

Comment:
Looking on the amount of code necessary for GPDR and COPPA compliance in `ExchangeService` I start to think that it might make sense to extract related logic into separate service. @rpanchyk what do you think?

Reply:
Yeah, i think it is a great candidate for new PR :-)

Reply:
Discussed offline and decided to do this refactoring separately.